libexec/rtld-elf/rtld_c18n.c

```diff
@@ -561,23 +563,28 @@ tramp_should_include(const Obj_Entry *reqobj, const struct tramp_data *data)
 	if (string_base_search(&uni_compart.trusts, sym) != -1)
 		return (false);
 
-	if (reqobj == NULL)
+	if (reqobj == NULL) {
+		assert(caller == RTLD_COMPART_ID);
 		return (true);
+	}
```

Comment:
```suggestion
```

This check is supplanted by the `plt != NULL` check below. And the logic afterwards should give us the right answer (i.e. `return (true);`), so there's no need for the special case here.